presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetTypeUtils.java

```diff
@@ -111,14 +109,18 @@ public static ColumnIO getArrayElementColumn(ColumnIO columnIO)
         return columnIO;
     }
 
-    public static Map<List<String>, RichColumnDescriptor> getDescriptors(MessageType fileSchema, MessageType requestedSchema)
+    public static Map<List<String>, RichColumnDescriptor> getDescriptors(MessageType fileSchema, MessageType requestedSchema, Map<parquet.schema.Type, HiveColumnHandle> typeColumns)
     {
         Map<List<String>, RichColumnDescriptor> descriptorsByPath = new HashMap<>();
         List<PrimitiveColumnIO> columns = getColumns(fileSchema, requestedSchema);
         for (String[] paths : fileSchema.getPaths()) {
             List<String> columnPath = Arrays.asList(paths);
             getDescriptor(columns, columnPath)
-                    .ifPresent(richColumnDescriptor -> descriptorsByPath.put(columnPath, richColumnDescriptor));
+                    .ifPresent(richColumnDescriptor -> {
```

Comment:
I think you can change this logic to pass the hive type to the rich column descriptor constructor instead of setting it via a setter. Please see my comment below for `RichColumnDescriptor`.